commit 9dec6f9c48242eec742c9475f32eeef29448701c
Author: Christian Lamparter <chunkeey@googlemail.com>
Date:   Fri Sep 17 22:07:59 2010 +0200

    carl9170: use rx chainmask from eeprom
    
    The eeprom provides a mask for all present rx chains.
    Why not use it instead of the generic initval default?
    
    Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

--- a/drivers/net/wireless/ath/carl9170/phy.c
+++ b/drivers/net/wireless/ath/carl9170/phy.c
@@ -532,8 +532,11 @@ static int carl9170_init_phy_from_eeprom
 	SET_VAL(AR9170_PHY_TPCRG1_PD_GAIN_2, val,
 		xpd2pd[m->xpdGain & 0xf] >> 2);
 	carl9170_regwrite(AR9170_PHY_REG_TPCRG1, val);
-	carl9170_regwrite_finish();
 
+	carl9170_regwrite(AR9170_PHY_REG_RX_CHAINMASK, ar->eeprom.rx_mask);
+	carl9170_regwrite(AR9170_PHY_REG_CAL_CHAINMASK, ar->eeprom.rx_mask);
+
+	carl9170_regwrite_finish();
 	return carl9170_regwrite_result();
 }
 
